cdp-docs/docs/program_overview/practitioner/guides/enterprise_modeling_guide/dbt_project_modeling_guide.md

```diff
@@ -0,0 +1,392 @@
+> Documentation needed.
+
+## dbt_project_name/data/
+> Documentation needed.
+
+## dbt_project_name/dbt_packages/
+> Documentation needed.
+
+## dbt_project_name/logs/
+> Documentation needed.
+
+## dbt_project_name/macros/
+> Documentation needed.
+
+## dbt_project_name/models/
+### Model Best Practices
+- The general naming convention for all models and fields should be singular.
+> - If you had a dimension that held attributes for a student in a given term, would you call it “dim_student_term”, “dim_student_terms”, or “dim_students_terms”?
+> - Simplicity – singular is shorter and doesn’t vary based on word (e.g., “ies”, “es”, “s”)
+> - Coding Principals – in coding, entities are always singular (e.g., Public Class Inventory)
+> - English Language Learners – pluralization is difficult, e.g., “inventory” vs “inventories”
+- Use CTEs.
+> A recognizable trait of dbt Lab's modeling style is their use of CTEs. We have adapted dbt Lab's use of CTEs. Commonly replacing subqueries with CTEs and utilizing CTEs to "import" models. New developers should review dbt's documentation on the topic of CTEs and existing CDP models.
+> - [Analytics Engineering Glossary: CTE in SQL](https://docs.getdbt.com/terms/cte)
+> - ['Import' CTEs](https://docs.getdbt.com/best-practices/how-we-style/2-how-we-style-our-sql#import-ctes) vs ['Functional' CTEs](https://docs.getdbt.com/best-practices/how-we-style/2-how-we-style-our-sql#functional-ctes)
+> - [Why dbt Labs (Fishtown) SQL style guide uses so many CTEs](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091)
+> - [Add CDP example models]
+- Developers should prioritize modular, reusable, concise, and universal modeling.
+> - Again, CTEs can be used to create modular and concise modeling.
+> - Intermediate models can be used to create modular, reusable, and concise models.
+> - Adhering to CDP's SQL style guide can create concise and universal models.
+
+### models/base/
+```
+└─── dbt_project_name
+     ├─── ...
+     ├─── models
+     │    └─── base
+               ├─── source1
+               │    ├─── _base_source1__sources.yml
+               │    ├─── _base_source1__models.yml
+               │    ├─── base_source1__model1.sql
+               │    ├─── ...
+               │    └─── base_source1__modeln.yml
+               ├─── ....
+               └─── sourcen
+     │    ├─── ...
+     ├─── ...
+
+```
+The base layer should be organized by source. Each base folder should contain a source yaml, model yaml, and base models.
+
+#### BEST PRACTICES
+- The base layer should only be used on a case by case basis.
+- The base layer should be the landing zone for semi-structure source data that needs to be parsed. (e.g. JSON data)
+- The base layer should be `ref()` in the staging layer.
+- The base layer should be used if there is a need for a model to retain the deleted records that are usually excluded in the staging layer. (e.g. `_fivetran_deleted = false`)
```

Comment:
I think we decided that we'd use `dbt-snapshots` rather than the `fivetran_deleted` column.